pkg/monitor/cluster/nodeconditions.go

```diff
@@ -26,24 +30,28 @@ func (mon *Monitor) emitNodeConditions(ctx context.Context) error {
 	mon.emitGauge("node.count", int64(len(ns.Items)), nil)
 
 	for _, n := range ns.Items {
+		nodeConditions := make(map[string]string)
+		isSpotInstance := mon.isSpotInstance(ctx, n)
+		nodeConditions["nodeName"] = n.Name
+		nodeConditions["spotInstance"] = strconv.FormatBool(isSpotInstance)
```

Comment:
can we do this only once per node AND only if there exists a "bad" status condition? This would move the logic down below lines 38.

Reply:
I've changed the impl to now only make two api calls in total. One to list nodes, and one to list machines.